### PR TITLE
Refactor/v1

### PR DESCRIPTION
**Pull request by jbartadev**
- Added `call` method for actor and task clients
- Added `tests` for call method on actor and task both
- Added private `_call` method to resource client

### CODE REVIEW

File: src/resource_clients/actor.js [hunk 1]
```diff
@@ -66,36 +66,23 @@ class ActorClient extends ResourceClient {
     async call(options = {}) {
         ow(options, ow.object.exactShape({
             contentType: ow.optional.string,
-            memoryMbytes: ow.optional.number,
-            timeoutSecs: ow.optional.number,
+            memory: ow.optional.number,
+            timeout: ow.optional.number,
```
Comment (mnmkng) [inline src/resource_clients/actor.js:70]: Let's keep the `Mbytes` and `Secs`. It's in line with our naming. We always add the units to number variables.

File: src/resource_clients/actor.js [hunk 2]
```diff
@@ -66,36 +66,23 @@ class ActorClient extends ResourceClient {
...
-            contentType,
-            waitForFinish: waitSecs,
-            memory: memoryMbytes,
-            build,
-            webhooks,
-            input,
-        };
+        const { timeout, ...callOptions } = options;
+
+        if (timeout >= 0) callOptions.timeout = timeout;
```
Comment (mnmkng) [inline src/resource_clients/actor.js:79]: I'm thinking, instead of this, we could validate that the `timeoutSecs` directly in `ow` to be non-negative.

Comment (mnmkng) [reply src/resource_clients/actor.js:79]: Same for `waitSecs` actually.

File: src/resource_clients/task.js [hunk 3]
```diff
@@ -53,34 +53,22 @@ class TaskClient extends ResourceClient {
 
     async call(options = {}) {
         ow(options, ow.object.exactShape({
-            memoryMbytes: ow.optional.number,
-            timeoutSecs: ow.optional.number,
+            memory: ow.optional.number,
+            timeout: ow.optional.number,
```
Comment (mnmkng) [inline src/resource_clients/task.js:57]: same here

File: test/tasks.test.js [hunk 4]
```diff
@@ -189,6 +189,56 @@ describe('Task methods', () => {
...
+            const body = { data };
+            const waitSecs = 1;
+
+            const query = {
+                timeout,
+                memory,
+                build,
+            };
+
+            mockServer.setResponse({ body }); // this is not used for the task.start()
```
Comment (mnmkng) [inline test/tasks.test.js:210]: actually, here it is used for `task.start()` too

**Review by mnmkng — COMMENTED** [review]
Looks good. Some minor improvements and it's good to go. Thanks a lot @jbartadev, it was pretty painful.

File: src/resource_clients/actor.js (hunk 1 above)
Comment (jbartadev) [reply src/resource_clients/actor.js:70]: Ok, make sense. Renamed.

File: src/resource_clients/actor.js (hunk 2 above)
Comment (jbartadev) [reply src/resource_clients/actor.js:79]: Added the non-negative validation, removed the condition 👍

File: src/resource_clients/task.js (hunk 3 above)
Comment (jbartadev) [reply src/resource_clients/task.js:57]: sure.

**Review by mnmkng — APPROVED** [review]

**Pull request merged**